rpc/service.go

```diff
@@ -18,32 +18,49 @@ func AttachService(manager jasper.Manager, s *grpc.Server) error {
 	return errors.WithStack(internal.AttachService(manager, s))
 }
 
-// StartService starts an RPC server with the specified address. If certFile
-// and keyFile are non-empty, the credentials will be read from the files to
-// start a TLS service; otherwise it will start an insecure service. The caller
-// is responsible for closing the connection using the returned
-// jasper.CloseFunc.
-func StartService(ctx context.Context, manager jasper.Manager, address net.Addr, certFile string, keyFile string) (jasper.CloseFunc, error) {
-	lis, err := net.Listen(address.Network(), address.String())
+// StartService starts an RPC server with the specified address addr around the
+// given manager. If creds is non-nil, the credentials will be used to establish
+// a secure TLS connection with clients; otherwise, it will start an insecure
+// service. The caller is responsible for closing the connection using the
+// return jasper.CloseFunc.
+func StartService(ctx context.Context, manager jasper.Manager, addr net.Addr, creds *Credentials) (jasper.CloseFunc, error) {
+	lis, err := net.Listen(addr.Network(), addr.String())
 	if err != nil {
-		return nil, errors.WithStack(err)
+		return nil, errors.Wrapf(err, "error listening on %s", addr.String())
 	}
 
-	var service *grpc.Server
-	if certFile != "" && keyFile != "" {
-		creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	opts := []grpc.ServerOption{}
+	if creds != nil {
+		tlsConf, err := creds.Resolve()
 		if err != nil {
-			return nil, errors.Wrapf(err, "could not get server credentials from cert file '%s' and key file '%s'", certFile, keyFile)
+			return nil, errors.Wrap(err, "error generating TLS config from server credentials")
 		}
-		service = grpc.NewServer(grpc.Creds(creds))
-	} else {
-		service = grpc.NewServer()
+		opts = append(opts, grpc.Creds(credentials.NewTLS(tlsConf)))
 	}
 
+	service := grpc.NewServer(opts...)
```

Comment:
Probably should add some logging interceptors here

Reply:
I think you forgot to push

Reply:
I added it [here](https://github.com/mongodb/jasper/pull/81/files#diff-75b6eb34296ad137de7f0652ab98e00bR36)?

Reply:
Weird, it wasn't loading